### PR TITLE
AsyncTimelockResource is the only option for Timelock

### PR DESCRIPTION
**Pull request by gokcanozakdag**
**Goals (and why)**:
Removing the option of registering LegacyTimelockService as a resource for timelock server - to reduce complexity / duplicate work.

**Where should we start reviewing?**:
`AsyncTimelockServicesCreator`

### CODE REVIEW

**Comment by gokcanozakdag** [comment]
@jeremyk-91 thanks for pointing these issues ^. I have updated the changelog and removed `AsyncLockConfiguration` with associated tests.

File: timelock-agent/src/main/java/com/palantir/timelock/config/TimeLockInstallConfiguration.java [hunk 1]
```diff
@@ -38,9 +38,4 @@
     default TsBoundPersisterConfiguration timestampBoundPersistence() {
         return ImmutablePaxosTsBoundPersisterConfiguration.builder().build();
     }
-
-    @Value.Default
-    default AsyncLockConfiguration asyncLock() {
-        return ImmutableAsyncLockConfiguration.builder().build();
-    }
```
[the remark this answers is not in the document]
Comment (j-baker) [reply timelock-agent/src/main/java/com/palantir/timelock/config/TimeLockInstallConfiguration.java]: instead, please can we add a `@JsonIgnoreProperties("asyncLock")` or similar that will achieve the same effect?

File: docs/source/release_notes/release-notes.rst [hunk 2]
```diff
@@ -56,6 +56,10 @@ develop
            Please be very careful if you are explicitly setting this to ``CheckAndSetCompatibility.SUPPORTED_DETAIL_ON_FAILURE``.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/3713>`__)
 
+    *    - |userbreak|
```
Comment (j-baker) [inline docs/source/release_notes/release-notes.rst]: by json-ignoring, we can avoid the userbreak piece

**Comment by jeremyk-91** [comment]
(discussed offline, confirmed correct with internal config-scoping tools)

**Pull request merged**